src/sth_database.js

```diff
@@ -10,8 +10,14 @@
 
   var sthConfig, sthLogger, sthHelper, connectionURL, eventSchema, aggregatedSchema;
 
-  var MAX_NAMESPACE_SIZE_IN_BYTES = 120,
-      MIN_HASH_SIZE_IN_BYTES = 20;
+  /**
+   * The maximumum namespace size in bytes has to be decreased to 113 bytes
```

Comment:
:D Probably me too... :p The point is that to apply the TTL using Mongo, a new index has to be created on the corresponding temporal property (see http://docs.mongodb.org/v2.6/tutorial/expire-data/ ). That index is created in the default `system.indexes` collection and it seems it is not possible to create that index for aggregated data collections if the hashing mechanism is activated and more than 113 bytes are used for the whole namespace (database + collection name) :) I cannot say much more than that... :D